rviz_simulator/src/camera_calibration_optimizer.cpp

```diff
@@ -0,0 +1,636 @@
+  double rotation_array[9];
+  ceres::AngleAxisToRotationMatrix(rodrigues_angle_axis.data(), rotation_array);  // column major
+  Eigen::Matrix3d rotation_matrix = Eigen::Map<Eigen::Matrix3d>(rotation_array);
+
+  // getting translation and converting to Eigen::Vector3d
+  std::array<double, 3> translation = { reference_T_Target[3], reference_T_Target[4], reference_T_Target[5] };
+  Eigen::Vector3d translation_vector = Eigen::Map<Eigen::Vector3d>(translation.data());
+
+  // combining rotation matrix and translation vector to Eigen::Affine3d
+  Eigen::Affine3d transform;
+  transform.translation() = translation_vector;
+  transform.linear() = rotation_matrix;
+
+  // inverting the transform
+  Eigen::Affine3d transform_inverse = transform.inverse();
+
+  // converting from rotation matrix back to rodriguez angle axis
+  ceres::RotationMatrixToAngleAxis(transform_inverse.rotation().data(), rodrigues_angle_axis.data());
+
+  // storing in an std::array
+  reference_T_target_inverse[0] = rodrigues_angle_axis[0];
+  reference_T_target_inverse[1] = rodrigues_angle_axis[1];
+  reference_T_target_inverse[2] = rodrigues_angle_axis[2];
+
+  reference_T_target_inverse[3] = transform_inverse.translation().x();
+  reference_T_target_inverse[4] = transform_inverse.translation().y();
+  reference_T_target_inverse[5] = transform_inverse.translation().z();
+
+  return reference_T_target_inverse;
+}
+
+std::array<double, 6> getReference_T_Target(const YAML::Node& node, std::string reference_T_target_name, bool inverse)
+{
+  std::array<double, 6> reference_T_target;
+
+  reference_T_target[0] = node[reference_T_target_name]["rotation"][0].as<double>();
+  reference_T_target[1] = node[reference_T_target_name]["rotation"][1].as<double>();
+  reference_T_target[2] = node[reference_T_target_name]["rotation"][2].as<double>();
+
+  reference_T_target[3] = node[reference_T_target_name]["translation"][0].as<double>();
+  reference_T_target[4] = node[reference_T_target_name]["translation"][1].as<double>();
+  reference_T_target[5] = node[reference_T_target_name]["translation"][2].as<double>();
+
+  if (inverse)
+  {
+    return reference_T_TargetInverse(reference_T_target);
+  }
+  else
+  {
+    return reference_T_target;
+  }
+}
+
+std::map<int, Target> CameraCalibrationOptimizer::getTargets(std::string targets_directory_path)
+{
+  YAML::Node targets_node = YAML::LoadFile(targets_directory_path);
+  if (!targets_node)
+  {
+    ROS_ERROR("Targets file read error.");
+  }
+
+  std::map<int, Target> targets;
+
+  // iterating through all targets
+  for (YAML::const_iterator target_iterator = targets_node["targets"].begin();
+       target_iterator != targets_node["targets"].end(); ++target_iterator)
+  {
+    const YAML::Node& target_node = *target_iterator;
+    Target target;
+
+    // targetID
+    target.targetID = target_node["targetID"].as<int>();
```

Comment:
With the change above: 
```suggestion
    target.targetID = target_iterator->operator[]("targetID").as<int>();
```
or
```suggestion
    target.targetID = (*target_iterator)["targetID"].as<int>();
```